torch/csrc/jit/passes/constant_propagation.cpp

```diff
@@ -71,18 +118,36 @@ void ConstantPropagation(Node* n, bool recurse) {
       return v->node()->kind() == prim::Constant;
     });
   bool supported_node = skip_list.count(n->kind()) == 0;
-  if (constant_inputs && supported_node) {
+  auto run_blocks = [&]() {
+    if (recurse) {
+      for (Block * block : n->blocks()) {
+        ConstantPropagation(block, recurse);
+      }
+    }
+  };
+  if (n->kind() == prim::If) {
+    run_blocks();
+    //inline node if we can, otherwise check for simplified outputs
+    if (constant_inputs) {
+      inlineIf(n);
+    } else {
+      removeExtraNodeOutputs(n);
```

Comment:
I think this needs to be done for prim::Loop nodes as well. An if statement that is constant prop inside the loop could also cause a loop carried dependency to no longer change.

Reply:
I have that commented as a todo below the run_blocks() code below